### DISCUSSION TITLE
Quarkus Camel Bom missing dependency

### DISCUSSION THREAD

**Discussion by tbkahuna48** [quarkusio/quarkus#36845] [Q&A]
Compiling camel quarkus code that previously compiled, get this error 
java.lang.ClassNotFoundException: org.apache.camel.component.vertx.common.VertxBufferConverterLoader

<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>org.kp.kpd.ee</groupId>
	<artifactId>ee-quarkus-parent-pom</artifactId>
	<version>1.0.5-SNAPSHOT</version>
	<packaging>pom</packaging>
	<properties>
		<quarkus.platform.artifactId>quarkus-bom</quarkus.platform.artifactId>
		<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
		<quarkus.platform.version>3.5.0</quarkus.platform.version>
		<compiler-plugin.version>3.11.0</compiler-plugin.version>
		<surefire-plugin.version>3.0.0</surefire-plugin.version>
		<failsafe-plugin.version>3.1.2</failsafe-plugin.version>
		<maven.deploy.plugin.version>3.0.0</maven.deploy.plugin.version>
		<maven.compiler.source>17</maven.compiler.source>
		<maven.compiler.release>17</maven.compiler.release>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
		<ee.quarkus.kp.logger.version>1.0.0</ee.quarkus.kp.logger.version>
		<ee.quarkus.common.lib.version>1.0.2</ee.quarkus.common.lib.version>
		<ee.quarkus.message.handler.version>1.0.2-SNAPSHOT</ee.quarkus.message.handler.version>
		<quarkus.package.add-runner-suffix>false</quarkus.package.add-runner-suffix>
		<quarkus.jandex.version>3.1.2</quarkus.jandex.version>
		<rest.assured.version>5.3.2</rest.assured.version>
	</properties>
	<distributionManagement>
		<repository>
			<id>inhouse_release</id>
			<name>bluemix-lib-releases</name>
			<url>https://artifactory-fof.appl.kp.org/artifactory/inhouse_release</url>
		</repository>
		<snapshotRepository>
			<id>inhouse_snapshot</id>
			<name>bluemix-lib-snapshots</name>
			<url>
				https://artifactory-fof.appl.kp.org/artifactory/inhouse_snapshot</url>
		</snapshotRepository>
	</distributionManagement>
	<repositories>
		<repository>
			<id>inhouse_release</id>
			<name>releaseRepo</name>
			<url>https://artifactory-fof.appl.kp.org/artifactory/inhouse_release</url>
		</repository>
		<repository>
			<id>inhouse_snapshot</id>
			<name>snapShot</name>
			<url>
				https://artifactory-fof.appl.kp.org/artifactory/inhouse_snapshot</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</repository>
	</repositories>
	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>${quarkus.platform.group-id}</groupId>
				<artifactId>${quarkus.platform.artifactId}</artifactId>
				<version>${quarkus.platform.version}</version>
				<scope>import</scope>
				<type>pom</type>
			</dependency>
			
			<dependency>
				<groupId>jakarta.platform</groupId>
				<artifactId>jakarta.jakartaee-bom</artifactId>
				<version>10.0.0</version>
				<type>pom</type>
			</dependency>
			<dependency>
				<groupId>org.kp.kpd.ee</groupId>
				<artifactId>ee-quarkus-kp-logger</artifactId>
				<version>${ee.quarkus.kp.logger.version}</version>
			</dependency>
			<dependency>
				<groupId>org.kp.kpd.ee</groupId>
				<artifactId>ee-quarkus-common-lib</artifactId>
				<version>${ee.quarkus.common.lib.version}</version>
			</dependency>
			<dependency>
				<groupId>io.rest-assured</groupId>
				<artifactId>rest-assured</artifactId>
				<version>${rest.assured.version}</version>
				<scope>test</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>
</project>

<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.kp.kpd.ee</groupId>
		<artifactId>ee-quarkus-parent-pom</artifactId>
		<version>1.0.5-SNAPSHOT</version>
	</parent>
	<groupId>org.kp.kpd.ee</groupId>
	<artifactId>ee-quarkus-camel-parent-pom</artifactId>
	<version>1.0.2-SNAPSHOT</version>
	<packaging>pom</packaging>
	<properties>
		<camel.quarkus.platform.group-id>io.quarkus.platform</camel.quarkus.platform.group-id>
		<camel.quarkus.platform.artifact-id>quarkus-camel-bom</camel.quarkus.platform.artifact-id>
	</properties>
	<dependencyManagement>
		<dependencies>			
			<dependency>
				<groupId>${camel.quarkus.platform.group-id}</groupId>
				<artifactId>${camel.quarkus.platform.artifact-id}</artifactId>
				<version>${quarkus.platform.version}</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>
</project>

<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.kp.kpd.ee</groupId>
		<artifactId>ee-quarkus-camel-parent-pom</artifactId>
		<version>1.0.2-SNAPSHOT</version>
	</parent>
	
	<groupId>org.kp.kpd.ee</groupId>
	<artifactId>ee-quarkus-reference-camel-service</artifactId>
	<version>1.0.0-SNAPSHOT</version>

	<properties>
		<formatter-maven-plugin.version>2.23.0</formatter-maven-plugin.version>
		<impsort-maven-plugin.version>1.9.0</impsort-maven-plugin.version>
		<license-maven-plugin.version>4.2</license-maven-plugin.version>
	</properties>

	<dependencies>
		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-core</artifactId>
		</dependency>
		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-log</artifactId>
		</dependency>
		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-bean</artifactId>
		</dependency>
		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-direct</artifactId>
		</dependency>
		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-microprofile-health</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-smallrye-health</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-config-yaml</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-rest-client-reactive-jackson</artifactId>
		</dependency>

		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-resteasy-reactive</artifactId>
		</dependency>

		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-rest-client-reactive</artifactId>
		</dependency>

		<dependency>
			<groupId>org.apache.camel.quarkus</groupId>
			<artifactId>camel-quarkus-microprofile-fault-tolerance</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-arc</artifactId>
		</dependency>
		<dependency>
			<groupId>io.quarkus</groupId>
			<artifactId>quarkus-junit5</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.rest-assured</groupId>
			<artifactId>rest-assured</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>
	<build>
		<plugins>
			<plugin>
				<groupId>${quarkus.platform.group-id}</groupId>
				<artifactId>quarkus-maven-plugin</artifactId>
				<version>${quarkus.platform.version}</version>
				<extensions>true</extensions>
				<executions>
					<execution>
						<goals>
							<goal>build</goal>
							<goal>generate-code</goal>
							<goal>generate-code-tests</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>${compiler-plugin.version}</version>
				<configuration>
					<compilerArgs>
						<arg>-parameters</arg>
					</compilerArgs>
				</configuration>
			</plugin>
			<plugin>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>${surefire-plugin.version}</version>
				<configuration>
					<systemPropertyVariables>
						<java.util.logging.manager>
							org.jboss.logmanager.LogManager</java.util.logging.manager>
						<maven.home>${maven.home}</maven.home>
					</systemPropertyVariables>
				</configuration>
			</plugin>
		</plugins>
	</build>

</project>

**Comment by maxandersen** [comment]
Looks like partial paste? 

Can you show the stscktrace, logs or even reproducer ?

**Comment by tbkahuna48** [comment]
Resolved , thanks.

From: Max Rydahl Andersen ***@***.***>
Date: Thursday, November 2, 2023 at 2:22 PM
To: quarkusio/quarkus ***@***.***>
Cc: Bryan Mclane ***@***.***>, Author ***@***.***>
Subject: Re: [quarkusio/quarkus] Quarkus Camel Bom missing dependency (Discussion #36845)

Caution: This email came from outside Kaiser Permanente. Do not open attachments or click on links if you do not recognize the sender.

________________________________

Looks like partial paste?

Can you show the stscktrace, logs or even reproducer ?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/quarkusio/quarkus/discussions/36845*discussioncomment-7461262__;Iw!!BZ50a36bapWJ!tZr-Ayioz7L6LRa3pMyf_EfZwRJyJwJZ9NKVNL48of1Jzn4wZA8KjpUWNLgr1uUIye8HwMHF2Xe68f0X9WbDMVH_yQA$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AJOLRWAMNLX3HSFCWYZ5EJ3YCQFKTAVCNFSM6AAAAAA63ID25CVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TINRRGI3DE__;!!BZ50a36bapWJ!tZr-Ayioz7L6LRa3pMyf_EfZwRJyJwJZ9NKVNL48of1Jzn4wZA8KjpUWNLgr1uUIye8HwMHF2Xe68f0X9WbD9zVXKic$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
NOTICE TO RECIPIENT:  If you are not the intended recipient of this e-mail, you are prohibited from sharing, copying, or otherwise using or disclosing its contents.  If you have received this e-mail in error, please notify the sender immediately by reply e-mail and permanently delete this e-mail and any attachments without reading, forwarding or saving them. v.173.295  Thank you.
